src/components/product/ProductGrid.tsx

```diff
@@ -30,9 +32,13 @@ export const ProductGrid: React.FC = () => {
     return (
         <div
             data-testid='product-grid'
-            className='flex flex-wrap justify-center gap-4 md:justify-start'
+            className='grid grid-cols-1 justify-center gap-6 sm:grid-cols-2 md:justify-start lg:grid-cols-3 xxl:grid-cols-4'
```

Comment:
🟡  This is a great improvement! But I was thinking, to address the card width issue, we could try a more flexible approach. What do you think about creating a responsive grid that automatically adjusts the number of columns based on the available space? Something like grid-cols-[repeat(auto-fit, minmax(250px, 1fr))] might work well here?

Reply:
that's a great idea, I'll try that. I want to avoid the case where there is some blank space because we fixed the card widths, but obviously they cannot stretch this much on bigger screens. Your solution should work well.